From 92f252f8a83fe2009a7648fc0f8371d9e6c80e3e Mon Sep 17 00:00:00 2001
From: Frank Wang <frank.wang@rock-chips.com>
Date: Wed, 15 Sep 2021 14:57:57 +0800
Subject: [PATCH 06/30] phy: rockchip: inno-usb2: fix rk3308bs tuning

This fix RK3308BS usbphy port 0/1 grf register write without mask bits.

Fixes: 074140216c20 ("phy: rockchip: inno-usb2: support rk3308bs tuning")
Signed-off-by: Frank Wang <frank.wang@rock-chips.com>
Change-Id: Iaa517a7a1dcfa1ab278726c14463295b88bb42b5
(cherry picked from commit 1719f06102918b62519b22796cd7775d443dadfb)
Reviewed-on: https://10.10.10.29/c/rk/kernel-stable/+/135262
Reviewed-by: Jianqun Xu <jay.xu@rock-chips.com>
Reviewed-by: Lin JIan Hua <linjh@rock-chips.com>
Reviewed-by: Hans Yang <yhx@rock-chips.com>
Tested-by: Hans Yang <yhx@rock-chips.com>
---
 drivers/phy/rockchip/phy-rockchip-inno-usb2.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/phy/rockchip/phy-rockchip-inno-usb2.c b/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
index 35ab9ec..b905351 100644
--- a/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
+++ b/drivers/phy/rockchip/phy-rockchip-inno-usb2.c
@@ -1899,17 +1899,17 @@ static int rk3308_usb2phy_tuning(struct rockchip_usb2phy *rphy)
 
 	if (soc_is_rk3308bs()) {
 		/* Turn off differential reciver in suspend mode */
-		ret = regmap_write(rphy->grf, 0x30, 0 | BIT(2) << 16);
+		ret = regmap_update_bits(rphy->grf, 0x30, BIT(2), 0);
 		if (ret)
 			return ret;
 
 		/* Enable otg port pre-emphasis during non-chirp phase */
-		ret = regmap_write(rphy->grf, 0, BIT(2) | GENMASK(2, 0) << 16);
+		ret = regmap_update_bits(rphy->grf, 0, GENMASK(2, 0), BIT(2));
 		if (ret)
 			return ret;
 
 		/* Enable host port pre-emphasis during non-chirp phase */
-		ret = regmap_write(rphy->grf, 0x400, BIT(2) | GENMASK(2, 0) << 16);
+		ret = regmap_update_bits(rphy->grf, 0x400, GENMASK(2, 0), BIT(2));
 		if (ret)
 			return ret;
 	} else {
-- 
2.7.4

